jsonschema-core.xml

```diff
@@ -135,13 +135,66 @@
                 data in several ways.
             </t>
             <t>
-                JSON Schema uses keywords to assert constraints on JSON instances or annotate those
-                instances with additional information.  Additional keywords are used to apply
-                assertions and annotations to more complex JSON data structures, or based on
-                some sort of condition.
+                A JSON Schema document is comprised of a set of keywords,
+                which are specified as properties in a JSON object.
+                The name of the keyword is used as the property name,
+                and any arguments to the keyword are provided as the value.
+                Keywords can be classified by their functions.
+                The most important classes are validation, annotation, and applicator keywords.
             </t>
             <t>
-                To facilitate re-use, keywords can be organized into vocabularies. A vocabulary
+                "Validation keywords" form the basis of JSON Schema.
+                Given a schema and an instance, the instance is "valid" if all of the keywords in
+                the schema are valid against the instance.
+                Validation keywords may be used to describe a set of JSON documents.
+                Without any validation keywords, the set of all valid instances is the set of all valid JSON documents.
+                Typically, adding validation keywords creates a subset.
+                Keywords are typically defined to not be redundant with other keywords:
```

Comment:
I don't know if redundancy is the right word for this (or whether putting it this early is needed at all). I'd probably not mention this level of detail at this point, it's more philosophical than really an overview, and it's also not immediately clear to me just how true it is, there's lots of keywords that "shrink" the set of valid instances across different types at once, like `const` or `enum`, or keywords added in a `not` schema, or ...

Reply:
The phrasing can be played with, but I think it's important to point out that `minLength` doesn't impact numbers and there's a good, predictable reason for that.

Reply:
To me it's again not very important to point out at this point, but if we do so I don't have too much of an issue with the way we've described the concept so far, namely that keywords may apply to a specific type, and if so, generally will then ignore (consider valid) instances of other types.